packages/collapsible-side-nav/src/SideNavItem.tsx

```diff
@@ -0,0 +1,213 @@
+  }
+`;
+
+const sideNavItemWithGlyphStyle = css`
+  & > svg:first-child {
+    margin-right: 5px;
+  }
+`;
+
+const sideNavItemLinkStyle = css`
+  font-weight: normal;
+  color: ${uiColors.blue.base};
+`;
+
+const sideNavItemActiveStyle = css`
+  background-color: ${uiColors.green.light3};
+  color: ${uiColors.green.dark3};
+  font-weight: bold;
+
+  &:hover {
+    background-color: ${uiColors.green.light3};
+  }
+`;
+
+const sideNavItemNonGroupStyle = css`
+  margin-top: 4px;
+  font-weight: bold;
+`;
+
+const sideNavItemCollapsedStyle = css`
+  height: 0;
+  opacity: 0;
+  visibility: hidden;
+`;
+
+const sideNavItemWithGlyphCollapsedStyle = css`
+  margin-top: 0;
+  height: 40px;
+  opacity: 1;
+  visibility: visible;
+  border-top-width: 1px;
+  color: ${uiColors.green.dark2};
+
+  &:hover {
+    cursor: initial;
+  }
+`;
+
+type Props = {
+  className?: string;
+  path?: string;
+} & OneOf<{ glyph: GlyphElement; glyphVisibility?: GlyphVisibility }, {}> &
+  OneOf<
+    { children: string },
+    { 'aria-label': string; children: React.ReactNode }
+  > &
+  Omit<HTMLElementProps<'a'>, 'ref'>;
+
+const SideNavItem = React.forwardRef(function SideNavItem(
+  {
+    className,
+    children,
+    glyph,
+    glyphVisibility = GlyphVisibility.OnlyCollapsed,
+    path,
+    href,
+    onClick,
+    'aria-label': ariaLabel,
+    ...rest
+  }: Props,
+  refProp,
+) {
+  const { collapsed, hovered, currentPath } = useContext(SideNavContext);
+  const groupContextData = useContext(SideNavGroupContext);
+
+  useEffect(() => {
+    if (groupContextData === null) {
+      return;
+    }
+
+    const { addPath, removePath } = groupContextData;
+
+    if (path) {
+      addPath(path);
+
+      return () => removePath(path);
+    }
+  }, [groupContextData, path]);
+
+  const setRef = useCallback(
+    (ref: HTMLElement | null) => {
+      if (groupContextData === null) {
+        return;
+      }
+
+      const { removePath } = groupContextData;
+
+      // Remove the path from the item's group if it was unmounted
+      if (ref === null && path !== undefined) {
+        removePath(path);
+      }
+
+      if (typeof refProp === 'function') {
+        refProp(ref);
+      } else if (refProp) {
+        refProp.current = ref;
+      }
+    },
+    [groupContextData, path, refProp],
+  );
+
+  const isActive = path !== undefined && currentPath === path;
+  const showCollapsed = collapsed && !hovered;
+  const hasGlyph = glyph !== undefined;
+  const isLink = href !== undefined;
+
+  let showGlyph: boolean;
+
+  if (hasGlyph) {
+    switch (glyphVisibility) {
+      case GlyphVisibility.OnlyCollapsed:
+        showGlyph = showCollapsed;
+        break;
+      case GlyphVisibility.OnlyExpanded:
+        showGlyph = !showCollapsed;
+        break;
+      case GlyphVisibility.Visible:
+        showGlyph = true;
+        break;
+      default:
+        enforceExhaustive(glyphVisibility);
+    }
+  } else {
+    showGlyph = false;
+  }
+
+  return (
+    <a
```

Comment:
should we take an `as` prop here to support more types of navigation?

Reply:
I would want to see a use-case for it. Accepting `as` and using `Box` usually makes things more difficult on the development side.

Reply:
using `Link` for example

Reply:
Fair enough 👍